src/graphics/imPipeline.cpp

```diff
@@ -0,0 +1,64 @@
+#define GLM_ENABLE_EXPERIMENTAL
+#include <GL/glew.h>
+#include <GL/gl.h>
+#include <glm/gtx/transform.hpp>
+#include <glm/ext.hpp>
+#include <surface/imMipmap.h>
+#include "imPipeline.h"
+
+void imPipeline::initialize() {
+
+    m_vertexShaderId = glCreateShader(GL_VERTEX_SHADER);
+    glShaderSource(m_vertexShaderId, 1, const_cast<const GLchar **>(&m_vertexShader), NULL);
```

Comment:
The `const_cast` should not be necessary, since the pointed-to data is already const.